pkg/util/controllerfinder/controller_finder.go

```diff
@@ -120,9 +120,10 @@ func (r *ControllerFinder) GetExpectedScaleForPods(pods []*corev1.Pod) (int32, e
 		workload, err := r.GetScaleAndSelectorForRef(ref.APIVersion, ref.Kind, pod.Namespace, ref.Name, ref.UID)
 		if err != nil && !errors.IsNotFound(err) {
 			return 0, err
-		}
-		if workload != nil && workload.Metadata.DeletionTimestamp.IsZero() {
+		} else if workload != nil && workload.Metadata.DeletionTimestamp.IsZero() {
 			controllerScale[workload.UID] = workload.Scale
+		} else if workload != nil && !workload.Metadata.DeletionTimestamp.IsZero() {
```

Comment:
if workload is deleted, no controllerScale entry will exists, it seems it is equivalent as setting controllerScale[workload.UID] = 0